src/sources/soundsourceffmpeg.cpp

```diff
@@ -22,6 +25,70 @@ void initFFmpegLib() {
     avcodec_register_all();
 }
 
+// More than 2 channels are currently not supported
+const SINT kMaxChannelCount = 2;
+
+inline
+AVMediaType getMediaType(AVStream* pStream) {
+    return pStream->codec->codec_type;
+}
+
+AVStream* findNextAudioStream(AVFormatContext* pFormatCtx, AVStream* pStream = nullptr) {
```

Comment:
Correct, currently only the first audio stream is actually used. Could be 
simplified by considering only this special case.

On 08.09.2016 20:23, Daniel Schürmann wrote:
>
> In src/sources/soundsourceffmpeg.cpp 
> <https://github.com/mixxxdj/mixxx/pull/1003#discussion_r78061828>:
>
> > @@ -22,6 +25,70 @@ void initFFmpegLib() {
> >      avcodec_register_all();
> >  }
> >
> > +// More than 2 channels are currently not supported
> > +const SINT kMaxChannelCount = 2;
> > +
> > +inline
> > +AVMediaType getMediaType(AVStream* pStream) {
> > +    return pStream->codec->codec_type;
> > +}
> > +
> > +AVStream* findNextAudioStream(AVFormatContext* pFormatCtx, AVStream* pStream = nullptr) {
>
> it looks like pStream is never set.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/1003/files/14399cf62b5d5ee96a4d0d3a361af8ab165b2144#r78061828>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-aoNe_qK2Q6GpRcc2ajmhMaAAw8I6ks5qoFKagaJpZM4J2QNu>.
>